public/app/plugins/datasource/loki/components/LokiContextUi.tsx

```diff
@@ -345,6 +377,36 @@ export function LokiContextUi(props: LokiContextUiProps) {
               />
             </>
           )}
+          {logContextProvider.queryContainsValidPipelineStages(origQuery) && (
+            <InlineFieldRow className={styles.operationsToggle}>
+              <InlineField
+                label="Include LogQL pipeline operations"
+                tooltip={
+                  <RenderUserContentAsHTML
+                    content={renderMarkdown(
+                      "This will include LogQL operations such as `line_format` or `label_format`. It won't include line or label filter operations."
+                    )}
+                  />
+                }
+              >
+                <InlineSwitch
+                  value={includePipelineOperations}
+                  showLabel={true}
+                  transparent={true}
+                  onChange={(e) => {
+                    reportInteraction('grafana_explore_logs_loki_log_context_pipeline_toggled', {
+                      logRowUid: row.uid,
+                      action: e.currentTarget.checked ? 'enable' : 'disable',
+                    });
+                    store.set(SHOULD_INCLUDE_PIPELINE_OPERATIONS, e.currentTarget.checked);
+                    setIncludePipelineOperations(e.currentTarget.checked);
+                    // need to retrigger a query to run by calling `updateFilter`
+                    updateFilter(contextFilters.filter(({ enabled }) => enabled));
```

Comment:
I think we should have a new function, something like `updateIncludePipelineOperations` that would trigger runContextQuery`.

Reply:
I am okay with leaving this this way with a comment, but my preference would be not manually triggering this trough context filters.

Reply:
I added the `runContextQuery` as a prop to `LokiContextUi` and just run that in 5a69c2ad4a2075b7e1137d952ae17826cf43fe0e